tool/resources/org/antlr/v4/tool/templates/codegen/Go/Go.stg

```diff
@@ -1196,102 +1295,116 @@ func New<struct.name>(parser antlr.Parser, ctx antlr.ParserRuleContext) *<struct
+func (s *<struct.name>) Set<a.name; format="cap">(v <TokenLabelType()>) { s.<a.name> = v \}
+}; separator="\n\n">
+
+//AltLabelStructDecl tokenTypeDecls
+<struct.tokenTypeDecls:{a | 
+func (s *<struct.name>) Get<a.name; format="cap">() int { return s.<a.name> \}
+func (s *<struct.name>) Set<a.name; format="cap">(v int) { s.<a.name> = v \}
+}; separator="\n\n">
+
+//AltLabelStructDecl tokenListDecls
+<struct.tokenListDecls:{a | 
+func (s *<struct.name>) Get<a.name; format="cap">() []<TokenLabelType()> { return s.<a.name> \}
+func (s *<struct.name>) Set<a.name; format="cap">(v []<TokenLabelType()>) { s.<a.name> = v \}}
+; separator="\n\n">
+
+//AltLabelStructDecl ruleContextDecls
+<struct.ruleContextDecls:{a | 
+func (s *<struct.name>) Get<a.name; format="cap">() I<a.ctxName> { return s.<a.name> \}
+func (s *<struct.name>) Set<a.name; format="cap">(v I<a.ctxName>) { s.<a.name> = v \}
+}; separator="\n\n">
+
+//AltLabelStructDecl ruleContextListDecls
+<struct.ruleContextListDecls:{a | 
+func (s *<struct.name>) Get<a.name; format="cap">() []I<a.ctxName> { return s.<a.name> \}
+func (s *<struct.name>) Set<a.name; format="cap">(v []I<a.ctxName>) { s.<a.name> = v \}
+}; separator="\n\n">
+
+//AltLabelStructDecl attributeDecls
+<struct.attributeDecls:{a | 
+func (s *<struct.name>) Get<a.name; format="cap">() <a.type> { return s.<a.name> \}
+func (s *<struct.name>) Set<a.name; format="cap">(v <a.type>) { s.<a.name> = v \}
+}; separator="\n\n">
+
+//getRuleContext
 func (s *<struct.name>) GetRuleContext() antlr.RuleContext {
 	return s
 }
 <if(getters)>
 
+//getters
 <getters:{g | func (s *<struct.name>) <g>}; separator="\n\n">
 <endif>
 
 <if(dispatchMethods)>
 
 <dispatchMethods; separator="\n\n">
 <endif>
+
+//END AltLabelStructDecl
+
+
 >>
 
 ListenerDispatchMethod(method) ::= <<
-func (s *<struct.name>) <if(method.isEnter)>Enter<else>Exit<endif>Rule(listener antlr.ParseTreeListener) {
-	if listenerT, ok := listener.(<parser.grammarName>Listener); ok {
-		listenerT.<if(method.isEnter)>Enter<else>Exit<endif><struct.derivedFromName; format="cap">(s)
-	}
+<if(method.isEnter)>
+func (s *<struct.name>) EnterRule(listener antlr.ParseTreeListener) {
+    if listenerT, ok := listener.(<struct.derivedFromName; format="cap">ContextListener); ok {
+        listenerT.Enter<struct.derivedFromName; format="cap">(s)
+    }
+}
+<else>
+func (s *<struct.name>) ExitRule(listener antlr.ParseTreeListener) {
+    if listenerT, ok := listener.(<struct.derivedFromName; format="cap">ContextListener); ok {
+        listenerT.Exit<struct.derivedFromName; format="cap">(s)
+    }
 }
+<endif>
 >>
 
 VisitorDispatchMethod(method) ::= <<
-func (s *<struct.name>) Accept(visitor antlr.ParseTreeVisitor) interface{} {
-	switch t := visitor.(type) {
-	case <parser.grammarName>Visitor:
-		return t.Visit<struct.derivedFromName; format="cap">(s)
-
+func (s *<struct.name>) Accept(delegate antlr.ParseTreeVisitor) {
```

Comment:
as noted before Accept should only get called by the generated code. Please provide an example of using the returned interface{}.